src/vs/editor/contrib/colorPicker/browser/colorHoverParticipant.ts

```diff
@@ -178,9 +190,9 @@ async function _createColorHover(participant: ColorHoverParticipant | Standalone
 	}
 }
 
-function renderHoverParts(participant: ColorHoverParticipant | StandaloneColorPickerParticipant, editor: ICodeEditor, themeService: IThemeService, hoverParts: ColorHover[] | StandaloneColorPickerHover[], context: IEditorHoverRenderContext) {
+function renderHoverParts(participant: ColorHoverParticipant | StandaloneColorPickerParticipant, editor: ICodeEditor, themeService: IThemeService, hoverParts: ColorHover[] | StandaloneColorPickerHover[], context: IEditorHoverRenderContext): { disposables: IDisposable; colorPicker: ColorPickerWidget | undefined } {
```

Comment:
I think this should return a disposable object instead of `{ disposables: ..., object: ... }`, to simplify the code. (Also check all the callers that they properly dispose it)

Reply:
WIP draft PR that aims to do this:https://github.com/microsoft/vscode/pull/214482
the color picker code needs some refactoring in general